pkg/cmd/verify/backupstorage.go

```diff
@@ -0,0 +1,227 @@
+// NewVerifyBackupStorage perform checks against the local filesystem and compares the available storage bytes with the
+// estimated size as reported by EndpointStatus.
+func NewVerifyBackupStorage() *cobra.Command {
+	verifyBackupStorage := &verifyBackupStorage{}
+	cmd := &cobra.Command{
+		Use:   "backup-storage",
+		Short: "performs checks to ensure storage is adequate for backup state",
+		Run: func(cmd *cobra.Command, args []string) {
+			must := func(fn func(ctx context.Context) error) {}
+			must(verifyBackupStorage.Run)
+		},
+	}
+
+	verifyBackupStorage.AddFlags(cmd.Flags())
+	return cmd
+}
+
+var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+
+func (v *verifyBackupStorage) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&v.endpoints, "endpoints", defaultEndpoint, "Comma separated listed of targets to perform health checks against. Default https://localhost:2379")
+	fs.StringVar(&v.backupPath, "backup-path", defaultBackupPath, "path to verify storage requirements.")
+	fs.StringVar(&v.clientCertFile, "cert-file", defaultCertFilePath, "etcd client certificate file.")
+	fs.StringVar(&v.clientKeyFile, "key-file", defaultKeyFilePath, "etcd client key file.")
+	fs.StringVar(&v.clientCACertFile, "cacert-file", defaultCaCertFilePath, "etcd client CA certificate file.")
+}
+
+func (v *verifyBackupStorage) Run(ctx context.Context) error {
+	defer utilruntime.HandleCrash()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	// handle teardown
+	shutdownHandler := make(chan os.Signal, 2)
+	signal.Notify(shutdownHandler, shutdownSignals...)
+	go func() {
+		select {
+		case <-shutdownHandler:
+			klog.Infof("Received SIGTERM or SIGINT signal, shutting down.")
+			close(shutdownHandler)
+			cancel()
+		case <-ctx.Done():
+			klog.Infof("Context has been cancelled, shutting down.")
+			close(shutdownHandler)
+			cancel()
+		}
+	}()
+
+	waitDuration := 1 * time.Second
+	timeoutDuration := 10 * time.Second
+
+	// Perform reasonable retry on non-fatal errors
+	err := wait.Poll(
+		waitDuration,
+		timeoutDuration,
+		func() (bool, error) {
+			return v.verifyBackupStorage(ctx)
+		})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (v *verifyBackupStorage) verifyBackupStorage(ctx context.Context) (bool, error) {
+	tlsInfo := transport.TLSInfo{
+		CertFile:      v.clientCertFile,
+		KeyFile:       v.clientKeyFile,
+		TrustedCAFile: v.clientCACertFile,
+	}
+	cli, err := newETCD3Client(ctx, tlsInfo)
+	if err != nil {
+		klog.Warningf("failed to create client: %v", err)
+		return false, nil
+	}
+	defer cli.Close()
+
+	members, err := cli.MemberList(ctx)
+	if err != nil {
+		klog.Warningf("failed checking member list: %v", err)
+		return false, nil
+	}
+
+	dbSizeBytes, err := getBackupSizeBytes(ctx, cli, members.Members)
+	if err != nil {
+		klog.Warningf("failed checking backup size: %v", err)
+		return false, nil
+	}
+
+	fsAvailableBytes, err := getPathAvailableSpaceBytes(v.backupPath)
+	if err != nil {
+		return true, err
```

Comment:
Should this not be false as we ran into error?

Reply:
yeah error should supercede but reads funny